internal/provisioner/plain/controllers/bundleinstance_controller.go

```diff
@@ -103,12 +103,13 @@ func (r *BundleInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	b := &rukpakv1alpha1.Bundle{}
 	if err := r.Get(ctx, types.NamespacedName{Name: bi.Spec.BundleName}, b); err != nil {
-		if apierrors.IsNotFound(err) {
-			bi.Status.InstalledBundleName = ""
+		currentInstalledCondition := meta.FindStatusCondition(bi.Status.Conditions, "Installed")
+		if apierrors.IsNotFound(err) && currentInstalledCondition != nil && currentInstalledCondition.Status != "True" {
+			bi.Status.InstalledBundleName += ""
 		}
 		meta.SetStatusCondition(&bi.Status.Conditions, metav1.Condition{
-			Type:    "Installed",
-			Status:  metav1.ConditionFalse,
+			Type:    "HasValidBundle",
+			Status:  metav1.ConditionTrue,
```

Comment:
Shouldn't this be `"False"` since we failed to lookup the bundle?
```suggestion
			Status:  metav1.ConditionFalse,
```

Reply:
Yes, it should be. Good catch, I was looking at the Reason where I should've been looking at the Type.